src/main/kotlin/ink/abb/pogo/scraper/tasks/EvolvePokemon.kt

```diff
@@ -0,0 +1,80 @@
+/**
+ * Pokemon Go Bot  Copyright (C) 2016  PokemonGoBot-authors (see authors.md for more information)
+ * This program comes with ABSOLUTELY NO WARRANTY;
+ * This is free software, and you are welcome to redistribute it under certain conditions.
+ *
+ * For more information, refer to the LICENSE file in this repositories root directory
+ */
+
+package ink.abb.pogo.scraper.tasks
+
+import POGOProtos.Networking.Responses.ReleasePokemonResponseOuterClass
+import com.pokegoapi.api.pokemon.Pokemon
+import com.pokegoapi.api.pokemon.PokemonMetaRegistry
+import ink.abb.pogo.scraper.Bot
+import ink.abb.pogo.scraper.Context
+import ink.abb.pogo.scraper.Settings
+import ink.abb.pogo.scraper.Task
+import ink.abb.pogo.scraper.util.Log
+import ink.abb.pogo.scraper.util.pokemon.getIvPercentage
+
+class EvolvePokemon : Task {
+    override fun run(bot: Bot, ctx: Context, settings: Settings) {
+        //count the current stack of possible evolves
+        var countEvolveStack = 0
+        val groupedPokemonForCount = ctx.api.inventories.pokebank.pokemons.groupBy { it.pokemonId }
+        groupedPokemonForCount.forEach {
+            if (settings.evolveBeforeTransfer.contains(it.key)){
+                // Get pokemonFamily meta information
+                val pokemonMeta = PokemonMetaRegistry.getMeta(it.key)
+                var maxPossibleEvolves = bot.api.inventories.candyjar.getCandies(pokemonMeta.family) / pokemonMeta.candyToEvolve
+                // Add the minimum value, depending on which is the bottleneck, amount of candy, or pokemon of this type in pokebank:
+                countEvolveStack +=  Math.min(maxPossibleEvolves,it.value.count())
+                // Release pokemon we wont be able to evolve anyway, because of lack of candy:
+                it.value.sortedByDescending { it.cp }.forEach {
+                    maxPossibleEvolves--
+                    if (maxPossibleEvolves < 0) {
+                        releasePokemon(it,"Not enough candy to save for evolve",ctx)
+                    }
+                }
+
+            }
+        }
+        Log.yellow("Stack of pokemon ready to evolve: $countEvolveStack pokemons")
+
+        // use lucky egg if above evolve stack limit and evolve the whole stack
+        if (countEvolveStack >= settings.evolveStackLimit) {
+            val startingXP = ctx.api.playerProfile.stats.experience
+            Log.yellow("Using Lucky Egg before evolving stack of $countEvolveStack pokemons")
+            var resultLuckyEgg = ctx.api.inventories.itemBag.useLuckyEgg()
```

Comment:
please use `ctx.api.cachedInventories` and wrap in try/catch because it may fail